src/vstest.console/TestPlatformHelpers/TestRequestManager.cs

```diff
@@ -427,11 +427,16 @@ private bool UpdateRunSettingsIfRequired(string runsettingsXml, List<string> sou
                         runConfiguration.DesignMode :
                         this.commandLineOptions.IsDesignMode;
 
+                    // Add or update console logger.
                     if (!designMode)
                     {
                         AddOrUpdateConsoleLogger(document, runsettingsXml);
                         settingsUpdated = true;
                     }
+                    else
+                    {
+                        settingsUpdated = settingsUpdated || UpdateConsoleLoggerIfExists(document, runsettingsXml);
```

Comment:
Why do we want to enable Console logger in DesignMode?

Reply:
If user has added in runsettings, we are allowing console logger.